File: file-operations.js
const fs = require('fs');
const path = require('path');

// Define paths
const sourceDir = 'D:\\repos\\ferco-posta\\frontend\\src\\features\\dashboard';
const targetDir = 'D:\\repos\\ferco-posta\\frontend\\src\\features\\configuracion';
const jsSourceFile = path.join(sourceDir, 'Configuracion.jsx');
const cssSourceFile = path.join(sourceDir, 'Configuracion.css');
const jsTargetFile = path.join(targetDir, 'Configuracion.jsx');
const cssTargetFile = path.join(targetDir, 'Configuracion.css');

try {
  // 1. Create the directory
  console.log('1. Creating directory:', targetDir);
  if (!fs.existsSync(targetDir)) {
    fs.mkdirSync(targetDir, { recursive: true });
    console.log('   ✓ Directory created successfully');
  } else {
    console.log('   ✓ Directory already exists');
  }

  // 2 & 3. Read and write Configuracion.jsx
  console.log('\n2. Reading file:', jsSourceFile);
  const jsContent = fs.readFileSync(jsSourceFile, 'utf8');
  console.log('   ✓ File read successfully (' + jsContent.length + ' bytes)');
  
  console.log('\n3. Writing to:', jsTargetFile);
  fs.writeFileSync(jsTargetFile, jsContent, 'utf8');
  console.log('   ✓ File written successfully');

  // 4 & 5. Read and write Configuracion.css
  console.log('\n4. Reading file:', cssSourceFile);
  const cssContent = fs.readFileSync(cssSourceFile, 'utf8');
  console.log('   ✓ File read successfully (' + cssContent.length + ' bytes)');
  
  console.log('\n5. Writing to:', cssTargetFile);
  fs.writeFileSync(cssTargetFile, cssContent, 'utf8');
  console.log('   ✓ File written successfully');

  // 6. Verify both files exist
  console.log('\n6. Verifying files exist:');
  const jsExists = fs.existsSync(jsTargetFile);
  const cssExists = fs.existsSync(cssTargetFile);
  console.log('   Configuracion.jsx exists:', jsExists);
  console.log('   Configuracion.css exists:', cssExists);

  // List directory contents
  console.log('\nDirectory contents of:', targetDir);
  const files = fs.readdirSync(targetDir);
  files.forEach(file => {
    const filePath = path.join(targetDir, file);
    const stats = fs.statSync(filePath);
    console.log('   -', file, '(' + stats.size + ' bytes)');
  });

  console.log('\n✓ All operations completed successfully!');
} catch (error) {
  console.error('✗ Error:', error.message);
  process.exit(1);
}

File: file-operations.py
import os
import shutil

source_dir = r'D:\repos\ferco-posta\frontend\src\features\dashboard'
target_dir = r'D:\repos\ferco-posta\frontend\src\features\configuracion'

# 1. Create target directory
print(f'1. Creating directory: {target_dir}')
os.makedirs(target_dir, exist_ok=True)
print('   ✓ Directory created successfully')

# 2. Read and write Configuracion.jsx
js_source = os.path.join(source_dir, 'Configuracion.jsx')
js_target = os.path.join(target_dir, 'Configuracion.jsx')
print(f'\n2. Reading file: {js_source}')
with open(js_source, 'r', encoding='utf-8') as f:
    js_content = f.read()
print(f'   ✓ File read successfully ({len(js_content)} bytes)')

print(f'\n3. Writing to: {js_target}')
with open(js_target, 'w', encoding='utf-8') as f:
    f.write(js_content)
print('   ✓ File written successfully')

# 4. Read and write Configuracion.css
css_source = os.path.join(source_dir, 'Configuracion.css')
css_target = os.path.join(target_dir, 'Configuracion.css')
print(f'\n4. Reading file: {css_source}')
with open(css_source, 'r', encoding='utf-8') as f:
    css_content = f.read()
print(f'   ✓ File read successfully ({len(css_content)} bytes)')

print(f'\n5. Writing to: {css_target}')
with open(css_target, 'w', encoding='utf-8') as f:
    f.write(css_content)
print('   ✓ File written successfully')

# 6. Verify files exist
print('\n6. Verifying files exist:')
js_exists = os.path.exists(js_target)
css_exists = os.path.exists(css_target)
print(f'   Configuracion.jsx exists: {js_exists}')
print(f'   Configuracion.css exists: {css_exists}')

# List directory contents
print(f'\nDirectory contents of: {target_dir}')
files = os.listdir(target_dir)
for file in files:
    filepath = os.path.join(target_dir, file)
    size = os.path.getsize(filepath)
    print(f'   - {file} ({size} bytes)')

print('\n✓ All operations completed successfully!')
